### PR TITLE
test(storage): add random table test

### PR DESCRIPTION
**Pull request by DvirYo-starkware**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1816)

### CODE REVIEW

**Comment by dan-starkware** [comment]
*[`crates/papyrus_storage/src/db/table_types/test_utils.rs` line 205 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1816#-Nv6XtYf1no6eBl6kY2p:-Nv6XtYgE7p6mNqN8x-E:bejcgqp) ([raw file](https://github.com/starkware-libs/papyrus/blob/160b4f92d4a77e0feadfb57b32e80a5355c017cc/crates/papyrus_storage/src/db/table_types/test_utils.rs#L205)):*
> ```XML
>         debug!("iteration: {iter:?}");
>         let wtxn = writer.begin_rw_txn().unwrap();
>         let random_op = rng.gen_range(0..3);
> ```

Can this depend on the trait somehow?
For example, what happens when we add another operation such as append? 



_Code quote:_
```XML
let random_op = rng.gen_range(0..3);
```

**Review by dan-starkware — APPROVED** [review]
Please refrain from using `test` in test names. 

> Reviewed 3 of 4 files at r1, 3 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1816#-:-Nvjte5L6KV5P6xVFTFt:b18nz19)* status: all files reviewed, 1 unresolved discussion (waiting on @DvirYo-starkware)

**Review by dan-starkware — APPROVED** [review]
> Reviewed 2 of 5 files at r4, 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1816)* status: all files reviewed, 1 unresolved discussion (waiting on @DvirYo-starkware)

**Review by DvirYo-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1816)* status: all files reviewed, 1 unresolved discussion

*[a discussion](https://reviewable.io/reviews/starkware-libs/papyrus/1816#-NvkLiac4F0951TLn8pL:-NvkLiac4F0951TLn8pM:b-29mp1f) (no related file):*
1. I don't see your comments in reviewable, so I changed what I remember we talked about
2. There are several changes because of rebase (mainly imports)
3. the constants were changed to include more options now locally; this test runs ~3 seconds

___

**Review by dan-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r6, 2 of 4 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1816)* status: 4 of 6 files reviewed, 2 unresolved discussions (waiting on @DvirYo-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/papyrus/1816#-NwIzMv2Er_NpkqjM3it:-NwIzMv2Er_NpkqjM3iu:b4cjnhl) (no related file):*
@eranreshef-starkware PTAL at the CI part

___
___
*[`.github/workflows/ci.yml` line 243 at r9](https://reviewable.io/reviews/starkware-libs/papyrus/1816#-NwIzHLH-fi_5zcpD_b8:-NwIzHLH-fi_5zcpD_b9:bfh4cw2) ([raw file](https://github.com/starkware-libs/papyrus/blob/d92a92c23aa87a8899b8e6ce15af60945f284cb2/.github/workflows/ci.yml#L243)):*
> ```YAML
>       - uses: dtolnay/rust-toolchain@stable
>       - run: DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker build -f papyrus_utilities.Dockerfile .
>     
> ```

Just a new line



_Code quote:_
```YAML
···
```
___
*[`crates/papyrus_storage/src/db/table_types/dup_sort_tables_test.rs` line 15 at r9](https://reviewable.io/reviews/starkware-libs/papyrus/1816#-NwIzdrc8czNHUDU7if9:-NwIzdrc8czNHUDU7ifA:b-n6d6dn) ([raw file](https://github.com/starkware-libs/papyrus/blob/d92a92c23aa87a8899b8e6ce15af60945f284cb2/crates/papyrus_storage/src/db/table_types/dup_sort_tables_test.rs#L15)):*
> ```XML
> #[test]
> fn common_prefix_compare_with_simple_table_random() {
>     println!("START");
> ```

Please remove

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -239,4 +239,23 @@ jobs:
...
+  random-table-test:
+    runs-on: ubuntu-latest
+    steps:
+      # run this job only if the path 'crates/papyrus_storage/src/db/**' is changed, because it takes around 2 minutes.
+      - uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            target_directory:
+              - 'crates/papyrus_storage/src/db/**'
```
Comment (eranreshef-starkware) [inline .github/workflows/ci.yml:253]: @DvirYo-starkware if you want some action to be executed only when a certain file/path changes, you can use the built-in syntax like in this example https://github.com/starkware-libs/papyrus/blob/2511b3a5a67ad6aef2b8c8aa35cd5a7a2de3d627/.github/workflows/helm-install.yml#L10

**Review by eranreshef-starkware — APPROVED** [review]
`*.yml` files look OK

**Review by dan-starkware — APPROVED** [review]
> Reviewed 2 of 4 files at r12, 2 of 2 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1816)* status: all files reviewed, 1 unresolved discussion (waiting on @DvirYo-starkware)

**Pull request merged**